### ISSUE TITLE
Implement grace period

### ISSUE THREAD

**Issue by zond** [zond/diplicity#166]
See https://docs.google.com/document/d/1HpXVSdC-0aFTFSW2HqFM_LijGHYuYhpx3DQ_Br50PZo/edit#

**Comment by zond** [comment]
I have a new and simplified suggestion for this:

1. `Grace period in minutes` option for games.
2. If deadline is reached then order input and chat is disabled and a message is sent out explaining who hasn't committed.
3. If deadline + grace period is reached then phase resolves.

Is this good enough? @JorenC, @tttppp, @johnpooch

**Comment by JorenC** [comment]
Sounds like a plan!

**Comment by tttppp** [comment]
I think you need to flesh out step 2 a bit. Currently this sounds like we're implementing a public shaming for the NMR player, as "the order input and chat is disabled". Presumably the NMR player(s) can still enter moves

Why is the chat disabled? I imagine players will want to ask questions like "Why hasn't the phase resolved?" "How do I enter my moves?" Etc.

Is this only for GMed games?

**Comment by zond** [comment]
Hm, I'm thinking if we don't disable anything this is just longer phases with a reminder and some public shaming?

**Comment by tttppp** [comment]
I agree - I thought it just looked like a reminder feature. The original document had an idea that each player got given a certain number of 'lives' - that would be different. Or if this was a GM feature then it would provide some time to investigate, and maybe replace a player.

**Comment by JorenC** [comment]
It might be a GM feature where no one can do anything but it just gives the
GM time.

However, it might be good to have it in public games as well, with number
of lives (default like 1-2 or the game drags out)
